bundles/org.openhab.binding.melcloud/src/main/java/org/openhab/binding/melcloud/internal/api/json/ListDevicesResponse.java

```diff
@@ -0,0 +1,445 @@
+/**
+ * Copyright (c) 2010-2019 Contributors to the openHAB project
+ *
+ * See the NOTICE file(s) distributed with this work for additional
+ * information.
+ *
+ * This program and the accompanying materials are made available under the
+ * terms of the Eclipse Public License 2.0 which is available at
+ * http://www.eclipse.org/legal/epl-2.0
+ *
+ * SPDX-License-Identifier: EPL-2.0
+ */
+package org.openhab.binding.melcloud.internal.api.json;
+
+import com.google.gson.annotations.Expose;
+import com.google.gson.annotations.SerializedName;
+
+/**
+ * The {@link ListDevicesResponse} is responsible of JSON data For MELCloud API
+ * Response of Devices List.
+ * Generated with jsonschema2pojo
+ *
+ * @author Luca Calcaterra - Initial contribution
+ */
+public class ListDevicesResponse {
+
+    @Expose
+    private Integer iD;
+
+    @Expose
+    private String name;
+
+    @Expose
+    private String addressLine1;
+
+    @Expose
+    private Object addressLine2;
+
+    @Expose
+    private String city;
+
+    @Expose
+    private String postcode;
+
+    @Expose
+    private Double latitude;
+
+    @Expose
+    private Double longitude;
+
+    @Expose
+    private Object district;
+
+    @Expose
+    private Boolean fPDefined;
+
+    @Expose
+    private Boolean fPEnabled;
+
+    @Expose
+    private Integer fPMinTemperature;
+
+    @Expose
+    private Integer fPMaxTemperature;
+
+    @Expose
+    private Boolean hMDefined;
+
+    @Expose
+    private Boolean hMEnabled;
+
+    @Expose
+    private Object hMStartDate;
+
+    @Expose
+    private Object hMEndDate;
+
+    @Expose
+    private Integer buildingType;
+
+    @Expose
+    private Integer propertyType;
+
+    @Expose
+    private String dateBuilt;
+
+    @Expose
+    private Boolean hasGasSupply;
+
+    @Expose
+    private String locationLookupDate;
+
+    @Expose
+    private Integer country;
+
+    @Expose
+    private Integer timeZoneContinent;
+
+    @Expose
+    private Integer timeZoneCity;
+
+    @Expose
+    private Integer timeZone;
+
+    @Expose
+    private Integer location;
+
+    @Expose
+    private Boolean coolingDisabled;
+
+    @Expose
+    private Boolean expanded;
+
+    @Expose
+    private Structure structure;
+
+    @Expose
+    private Integer accessLevel;
+
+    @Expose
+    private Boolean directAccess;
+
+    @Expose
+    private Integer minTemperature;
+
+    @Expose
+    private Integer maxTemperature;
+
+    @Expose
+    private Object owner;
+
+    @Expose
+    private String endDate;
+
+    @SerializedName("iDateBuilt")
```

Comment:
This looks the same or is the problem that this field does not use underscores?

Reply:
MELCloud expects `iDateBuilt` but since we have set `setFieldNamingPolicy(FieldNamingPolicy.UPPER_CAMEL_CASE)` it would be translated to `IDateBuilt` without the `@SerializedName` annotation.